pkg/auth/ldaputil/client.go

```diff
@@ -78,3 +84,24 @@ func (l *LDAPClientConfig) Connect() (*ldap.Conn, error) {
 		return nil, fmt.Errorf("unsupported scheme %q", l.Scheme)
 	}
 }
+
+// Connect returns an established LDAP connection to which a bind has occurred if a bind DN and
+// password were given. If no error is returned, the caller is responsible for closing the connection
+func (l *LDAPClientConfig) Connect() (*ldap.Conn, error) {
```

Comment:
how many places are you establishing an LDAP connection? I'd expect about one for the group sync and one for the password auth...